Fails to build with Boost 1.70 (Beta1)

Regressed by boostorg/asio@a72fbb0b867f:
```c++
src/cli/tls_http_server.cpp:438:31: error: no member named 'get_io_service' in
      'boost::asio::basic_socket_acceptor<boost::asio::ip::tcp, boost::asio::executor>'
                   m_acceptor.get_io_service(),
                   ~~~~~~~~~~ ^
src/cli/tls_proxy.cpp:373:31: error: no member named 'get_io_service' in
      'boost::asio::basic_socket_acceptor<boost::asio::ip::tcp, boost::asio::executor>'
                   m_acceptor.get_io_service(),
                   ~~~~~~~~~~ ^
```

--- a/src/cli/tls_http_server.cpp
+++ b/src/cli/tls_http_server.cpp
@@ -41,6 +41,12 @@
 
 #include "credentials.h"
 
+#if BOOST_VERSION >= 107000
+#define GET_IO_SERVICE(s) ((boost::asio::io_context&)(s).get_executor().context())
+#else
+#define GET_IO_SERVICE(s) ((s).get_io_service())
+#endif
+
 namespace Botan_CLI {
 
 namespace {
@@ -435,7 +441,7 @@ class TLS_Asio_HTTP_Server final
       session::pointer make_session()
          {
          return session::create(
-                   m_acceptor.get_io_service(),
+                   GET_IO_SERVICE(m_acceptor),
                    m_session_manager,
                    m_creds,
                    m_policy);

--- a/src/cli/tls_proxy.cpp
+++ b/src/cli/tls_proxy.cpp
@@ -33,6 +33,12 @@
 
 #include "credentials.h"
 
+#if BOOST_VERSION >= 107000
+#define GET_IO_SERVICE(s) ((boost::asio::io_context&)(s).get_executor().context())
+#else
+#define GET_IO_SERVICE(s) ((s).get_io_service())
+#endif
+
 namespace Botan_CLI {
 
 namespace {
@@ -370,7 +376,7 @@ class tls_proxy_server final
       session::pointer make_session()
          {
          return session::create(
-                   m_acceptor.get_io_service(),
+                   GET_IO_SERVICE(m_acceptor),
                    m_session_manager,
                    m_creds,
                    m_policy,

